packages/plugins/src/SearchIndexPlugin.ts

```diff
@@ -9,18 +9,70 @@ type LeafNode = Node & {
   value: string;
 };
 
+interface IndexItem {
+  title: string;
+  route: string;
+  content?: string[];
+  [key: string]: string | string[];
+}
+
+interface SearchRelevancy {
+  includeScore?: boolean;
+  includeMatches?: boolean;
+  maxPatternLength?: number;
+  ignoreLocation?: boolean;
+  threshold?: number;
+  keys?: string[] | { name: string; weight: number }[];
+}
 interface SearchIndexPluginOptions {
   maxLineLength?: number;
   maxLineCount?: number;
   keys?: string[];
-  disabled?: boolean;
+  relevancy?: SearchRelevancy;
 }
 
 interface Optimization {
   content: string;
   title: string;
-  maxLineCount: number;
-  maxLineLength: number;
+  maxLineCount?: number;
+  maxLineLength?: number;
+}
+
+const defaultRelevancyOptions = {
+  includeScore: true,
+  includeMatches: true,
+  maxPatternLength: 240,
+  ignoreLocation: true,
+  threshold: 0.3
+};
+
+/**
+ * We need these keys to *always* be included in the search index.
+ * Without the title, there would be no way to display the search results.
+ * Without the route, there would be no way to navigate to the page.
+ */
+const requiredKeys = ['title', 'route'];
+
+/**
+ * If there are no keys specified in the plugin options, we'll use these
+ */
+const fallbackKeys = [...requiredKeys, 'content'];
+
+/**
+ * There are three scenarios for the "keys":
+ *
+ * 1. No keys are specified in the plugin options. In this case, we'll use the fallback keys.
+ * 2. The keys are specified as an array of strings. In this case, we'll add the required keys to the array.
+ * 3. The keys are specified as an array of objects. In this case, we can assume "weighting" is being applied
+ *    so we'll only return the keys that have been specified in the options.
+ *
+ * @param optionKeys array of keys specified in the plugin options
+ * @returns array of keys to be used in the search index
+ */
```

Comment:
Maybe we could move these comments above `key` property in the `SearchIndexPluginOptions` interface so the comment shows up in the IDE